src/layouts/ProfileLayout.tsx

```diff
@@ -45,69 +51,94 @@ function ProfileLayout({ children, wallet }: ProfileLayout): JSX.Element {
         />
         <link rel="icon" href="/favicon.ico" />
       </Head>
-      <Overview>
-        <Overview.Hero>
-          <Overview.Info
-            avatar={<Overview.Avatar src={wallet.previewImage} circle />}
-            title={<Overview.Title>{wallet.displayName}</Overview.Title>}
-          >
-            <Overview.Actions>
-              {isFollowingProfile ? (
+      <WalletIdentityProvider appName="Holaplex" appTwitter="@holaplex">
+        <Overview>
+          <Overview.Hero>
+            <Overview.Info
+              avatar={<Overview.Avatar src={wallet.previewImage} circle />}
+              title={<Overview.Title>{wallet.displayName}</Overview.Title>}
+            >
+              <Overview.Actions>
+                {isMyProfile && !wallet.profile?.handle && anchorWallet && (
+                  <ConnectTwitterButton
+                    address={new PublicKey(profileAddress)}
+                    connection={connection}
+                    wallet={anchorWallet}
+                    cluster={'mainnet-beta'}
+                    variant={'secondary'}
+                    style={{
+                      background: 'rgb(33,33,33)',
+                      height: '40px',
+                      width: '130px',
+                      borderRadius: '18px',
+                      padding: 16,
+                    }}
+                  />
+                )}
+
+                {isMyProfile ? null : isFollowingProfile ? (
+                  <Button
+                    loading={profileFollowingQuery.loading}
+                    size={ButtonSize.Small}
+                    type={ButtonType.Secondary}
+                  >
+                    {t('unfollow', { ns: 'common' })}
+                  </Button>
+                ) : (
+                  <Button
+                    loading={profileFollowingQuery.loading}
+                    icon={<PlusIcon width={14} height={14} />}
+                    size={ButtonSize.Small}
+                  >
+                    {t('follow', { ns: 'common' })}
+                  </Button>
+                )}
```

Comment:
I'm thinking we should go with a hook to gather all the info and then can render just 1 button. Move the use query for the connection into the hook.

```suggestion
  const followProfile = useFollowProfile(wallet, profile)
  
                {followProfile.followable && (
                  <Button
                    loading={followProfile.loading}
                    icon={followProfile.display.icon}
                    onClick={followProfile.onChange}
                    size={ButtonSize.Small}
                  >
                    {followProfile.display.label}
                  </Button>
                )}
```

Reply:
You can check out this PR for the follow feature. https://github.com/holaplex/dao-marketplace/pull/38

That uses the one Button approach, but like we talked about yesterday, moving the connection query to the button is not great for when we want to show many follow buttons in one view. 

Still, talking a bit with Damian and Belle on the whole Follow feature right now, which is why I marked it as draft.

Reply:
Because without Alpha, there is not too much utility in simply following someone.